### PR TITLE
migrations: Small refactor to runner package

### PR DESCRIPTION
**Pull request by efritz**
Pulled from #29831. This PR minimizes the diff between `main` and the experimental branch. This PR should make no behavioral changes.

### CODE REVIEW

File: internal/database/migration/runner/run.go [hunk 1]
```diff
@@ -44,6 +46,10 @@ func (r *Runner) Run(ctx context.Context, options Options) error {
 	})
 }
 
+// runSchema applies (or unapplies) the set of migrations required to fulfill the given operation. This
+// method will attempt to coordinate with with other concurrently running instances and may block while
```
Comment (camdencheek) [inline internal/database/migration/runner/run.go:50]:
```suggestion
// method will attempt to coordinate with other concurrently running instances and may block while
```

**Review by camdencheek — APPROVED** [review]

File: internal/database/migration/runner/run.go (hunk 1 above)
Comment (efritz) [reply internal/database/migration/runner/run.go:50]: Auto-merge: shakes-fist:

Comment (camdencheek) [reply internal/database/migration/runner/run.go:50]: how did you even automerge without a review?

Comment (efritz) [reply internal/database/migration/runner/run.go:50]: You approved but left comments :)

Comment (camdencheek) [reply internal/database/migration/runner/run.go:50]: Oh lol

**Pull request merged**